### PR TITLE
Better worn container display

### PR DESCRIPTION
**Pull request by Moltenhead**
#### Summary
Features "Quality of life improvement : add capacity essential data next to the worn containers within the inventory interface, add universal formatter to easily format common units `unit_to_string`"

#### Purpose of change
The actual interface needs you to check item by item to get essential capacity data for worn items (remaining volume, remaining mass, max item length).
Unit formating was quite tedious.

#### Describe the solution
Just displays a formatted string next to the worn container within the inventory interface such as `(remains $REMAINING_VOLUME, $REMAINING_MASS), max length $MAX_ITEM_LENGTH`. The data is only calculating unrestricted pockets handles any application's units settings and is displayed only if :
1. the item is worn
2. the item is a container
3. the container is containing at least one unrestricted pocket

Unrestricted pockets are pockets that are not flaged with any restrictions (such as sheath or ammo containers). the goal is to avoid flooding the interface while still having a faster peek at essential data.
Unit formating can now use `std::string unit_to_string(units::any, bool compact = false, bool remove_trailing_zeroes = false)` to quickly get a formatted value using the application units settings from a `units::volume`, `units::mass`, `units::length`.

#### Describe alternatives you've considered
The display could get activated / deactivated with a dedicated shortcut within the inventory interface.
It could also be extended to unrestricted pockets if needed (simply inverting the new accepted boolean argument within calculation methods `unrestricted_pockets_only` set to false by default, but using the true value for this feature).

#### Testing
* Tested multiple containers with various pockets setup (with / without restricted pockets).
* Filled then unfilled containers to check data update.
* Used insert actions to test the new display
* Displaying other related inventory ui panes
* Tested within multiple resolution, shrinking and extending windowed game

#### Additional context
New inventory :
bwd_global_interface
Container new display :
bwcd_common_display
Not affecting containers that only contains restricted pockets (weapon, ammo restricted, sheath, ...):
bwcd_restricted_pockets
Test on smaller display seems to be readable :
bwcd_small_display

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
Some screenshots would be helpful.

**Comment by Night-Pryanik** [comment]
I guess all this additional info won't be visible on smaller displays.

**Comment by Moltenhead** [comment]
There's a bug I've just found and I must correct, the display extends to other interfaces (wield, take off, wear, ...)

**Review by Moltenhead — COMMENTED** [review]
> I guess all this additional info won't be visible on smaller displays.

good point ... adding a shortcut to active / deactivate it could be useful in this case.

**Comment by Moltenhead** [comment]
> There's a bug I've just found and I must correct, the display extends to other interfaces (wield, take off, wear, ...)

Fixed with commit cf6f7be (may be a monkey patch - a better way to handle would be to detect and compare the active inventory pan, or even better apply the changes to the obtained text through cells get_text method)

**Comment by Moltenhead** [comment]
> > I guess all this additional info won't be visible on smaller displays.
> 
> good point ... adding a shortcut to activate / deactivate it could be useful in this case.

After some tests, smaller resolution won't hinder much the functionality, since the font size is not affected by the viewports.

File: src/inventory_ui.cpp [hunk 1]
```diff
@@ -316,7 +316,8 @@ std::string inventory_selector_preset::get_cell_text( const inventory_entry &ent
     } else if( entry.is_item() ) {
         std::string text = cells[cell_index].get_text( entry );
         const item &actual_item = *entry.locations.front();
-        if( entry.get_category_ptr()->get_id() == item_category_id( "ITEMS_WORN" ) &&
+        if( cell_index == 0 && !text.empty() &&
+            entry.get_category_ptr()->get_id() == item_category_id( "ITEMS_WORN" ) &&
```
Comment (Qrox) [inline src/inventory_ui.cpp:320]: Items are all displayed under the "ITEMS_WORN" category in the alternative inventory view (toggle with `;`). You might want to actually check that an item is worn by the player.

Comment (Moltenhead) [reply src/inventory_ui.cpp:320]: I'll test that thanks.

Comment (Moltenhead) [reply src/inventory_ui.cpp:320]: Thanks for the insight it would have took me a long time to figure this out on my own.

Comment (Moltenhead) [reply src/inventory_ui.cpp:320]: Fixed with 0cc290a

**Comment by actual-nh** [comment]
unit_to_string (and various altered versions): Why is `remove_trailing_zeroes` defaulting to false?

**Comment by Moltenhead** [comment]
> unit_to_string (and various altered versions): Why is `remove_trailing_zeroes` defaulting to false?

The game was made with trailing zeroes. Tests are made with trailing zeroes. If I don't default to false => the game breaks in lots of places.

**Comment by Moltenhead** [comment]
> > There's a bug I've just found and I must correct, the display extends to other interfaces (wield, take off, wear, ...)
> 
> Fixed with commit [cf6f7be](https://github.com/CleverRaven/Cataclysm-DDA/commit/cf6f7bec8cb53340ee55768cbcdea9fd37b586df) (may be a monkey patch - a better way to handle it would be to detect and compare the active inventory pan, or even better apply the changes to the obtained text through cells `cell_t::get_text` and not `inventory_selector_preset::get_cell_text` method)

Actualy, after digging the code and the way it is handled, the only "useful" option that I called here would be to detect the actual pan id, but it would only complexify this condition, the actual solution is working for everything and would also work if the inventory system is extended. It could become necessary if a new added inventory pan would need to display loads of data.

**Comment by Night-Pryanik** [comment]
@Moltenhead could you please resolve conflicts?

**Comment by Moltenhead** [comment]
> @Moltenhead could you please resolve conflicts?

I'll try to do that in the week. ;)

**Comment by Moltenhead** [comment]
Unrelated LGTM alerts - code from this pr is fine.

**Comment by I-am-Erk** [comment]
I am marking this project as "abandoned" but, since it is doing something we want, I have put it in an Abandoned PR list to hopefully be reinitiated at some point.

**Comment by Moltenhead** [comment]
> I am marking this project as "abandoned" but, since it is doing something we want, I have put it in an Abandoned PR list to hopefully be reinitiated at some point.

Why abandoned if ` it is doing something` you `want` and I was still resolving conflicts when asked ?
I really don't get how you are working. Everway I'll keep this one updated locally then.

**Comment by Night-Pryanik** [comment]
@Moltenhead could you please resolve conflicts?

**Comment by Moltenhead** [comment]
> @Moltenhead could you please resolve conflicts?

Ahahaha ! See ? This is not that difficult. x)
Welp I'll try to do that today if I can find some time.

**Comment by Shodan14** [comment]
Any way to toggle this on/off? It can get a bit cluttery, but I can see when it might also be useful.

**Comment by Moltenhead** [comment]
> Any way to toggle this on/off? It can get a bit cluttery, but I can see when it might also be useful.

not with the actual implementation.
Though it only needs to get  wrapped or safe guarded by a state to get displayed or not. So adding this plus a way to alter ui state would handle it's activity. I don't have enough time to add that atm, but it's a possibility.

**Pull request merged**